lib/fluent/plugin/kubernetes_metadata_watch_pods.rb

```diff
@@ -42,11 +54,9 @@ def start_pod_watch
             if cached
               @cache[cache_key] = parse_pod_metadata(notice.object)
               @stats.bump(:pod_cache_watch_updates)
-            elsif ENV['K8S_NODE_NAME'] == notice.object['spec']['nodeName'] then
+            else
               @cache[cache_key] = parse_pod_metadata(notice.object)
               @stats.bump(:pod_cache_host_updates)
-            else
-              @stats.bump(:pod_cache_watch_misses)
```

Comment:
Can we guarantee we will NEVER have a cache miss?

Reply:
We can't, good catch. Originally I assumed that K8S_NODE_NAME was always set, but then I learned it isn't. I've restored the code and the test. Sorry about that!

Reply:
On second thought, I think this code is redundant indeed. If K8S_NODE_NAME is set, there should be do misses due to filtering in the API. If it's not set, then the elsif block won't run. Your call whether to leave it or not.